libbeat/logp/core.go

```diff
@@ -196,6 +196,13 @@ func makeOptions(cfg Config) []zap.Option {
 	if cfg.development {
 		options = append(options, zap.Development())
 	}
+	if cfg.ECSEnabled {
+		fields := []zap.Field{zap.String("service.name", cfg.Beat)}
+		if cfg.Files.Name != "" {
```

Comment:
I see why this works, but it _looks_ wrong to me since we should add `event.dataset` for non-file logging output. How about we introduce a new function in this package like

```go
// logFilename returns the base filename to which logs will be written for
// the "files" log output. If another log output is used, or `logging.files.name`
// is unspecified, then the beat name will be returned.
func logFilename(cfg Config) string {
    name := cfg.Beat
    if cfg.Files.Name != "" {
        name = cfg.Files.Name
    }
    return name
}
```

Then you can use that here and in `makeFileOutput`, and revert the `configure.Logging`.

Reply:
> we should add event.dataset for non-file logging output

I don't get that. If the output is eg. stderr what should be the value of `event.dataset`?

Reply:
See https://github.com/elastic/beats/issues/25997 and https://github.com/elastic/ecs-logging/blob/9507c12f7523741cc06f84a2686d580e50f3465f/spec/spec.json#L79-L95

> If the user manually configures the service name, the logging library should set `event.dataset=${service.name}.log` if not explicitly configured otherwise.